libcaf_test/caf/test/test.test.cpp

```diff
@@ -40,6 +41,51 @@ TEST("tests can contain different types of checks") {
   info("this test had {} checks", rep.test_stats().total());
 }
 
+#ifdef CAF_ENABLE_EXCEPTIONS
+
+TEST("tests fail when requirement errors occur") {
+  auto& rep = caf::test::reporter::instance();
+  SECTION("require_eq fails when lhs != rhs") {
+    should_fail_with_exception<caf::test::requirement_failed>(
```

Comment:
```suggestion
    should_fail_with_exception<requirement_failed>(
```

Same for the others.